packages/cli/openapi-parser/src/v3/extensions/extensions.ts

```diff
@@ -1,41 +1,75 @@
-/**
- * Used to detect if an endpoint has an async version of it
- * Example usage:
- *   paths:
- *    /path/to/my/endpoint:
- *      x-fern-async-config:
- *        discriminant:
- *          type: header
- *          name: X-Header-Name
- *          value: async
- *        response-status-code: 202
- **/
-export const X_FERN_ASYNC_CONFIG = "x-fern-async-config";
+import { Values } from "@fern-api/core-utils";
 
-/**
- * Used to create veriables in the fern definition
- * Example usage:
- * x-fern-sdk-variables:
- *   appName:
- *     type: string
- * paths:
- *   /path/to/my/endpoint/{id}:
- *     parameters:
- *       - name: id
- *         in: path
- *         type: string
- *         x-fern-sdk-variable: appName
- */
-export const X_FERN_SDK_VARIABLES = "x-fern-sdk-variables";
-export const X_FERN_SDK_VARIABLE = "x-fern-sdk-variable";
+export const FernOpenAPIExtension = {
+    SDK_METHOD_NAME: "x-fern-sdk-method-name",
+    SDK_GROUP_NAME: "x-fern-sdk-group-name",
+    STREAMING: "x-fern-streaming",
+    REQUEST_NAME_V1: "x-request-name",
+    REQUEST_NAME_V2: "x-fern-request-name",
+    TYPE_NAME: "x-fern-type-name",
```

Comment:
If you want to add a type to an inlined schema or override an existing type name in OpenAPI, use `x-fern-type-name`

```
components:
  schemas:
     BadName: 
        x-fern-type-name: GoodName # <---- GoodName will be used in the SDK when generating Code
        type: object
        properties: 
           baz: string
```